inginious/frontend/plugins/grader_generator/static/js/grader_generator.js

```diff
@@ -166,3 +167,51 @@ function studio_update_container_name()
     try{ $("#multilang_grader_form")[0].style.display = "block"; } catch {};
   }
 }
+
+// Match test cases
+
+function read_files_and_match(){
+  // This function reads all the files on the tab "Task files" and 
+  // matches to test cases
+  $.get('/api/grader_generator/test_file_api', {
+    course_id: courseId,
+    task_id: taskId
+}, function(files) {
+    // Pass the file info to JSON for comparison
+    var json_files = [];
+    for(var ind = 0; ind < files.length; ind++)
+      json_files.push(JSON.stringify(files[ind]));
+
+    // Check the current test cases for no repetitions
+    var test_cases = []
+    for(ind = 0; ind < grader_test_cases_count; ind++)
+      test_cases.push($("#grader_test_cases_"+ ind + "_input_file").val());
+
+    $.each(files, function(index, file) {
+      if (file.is_directory || test_cases.includes(file.complete_name)) {
```

Comment:
I think we should check repeated entries instead of repeated files, as a file can be used for another test case.

Reply:
No, It cannot be used for other test case. Try to save a task with two test cases that used the same file for the input